### PR TITLE
HPCC-21294 Ensure WsFILEIO handles newer Dropzone config structure

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review

**Comment by wangkx** [comment]
@rpastrana do you know why the Smoketest failed?

**Comment by richardkchapman** [comment]
These failures are normal on 6.4.x (the regression suite requires 7.x to work properly)

File: esp/services/ws_fileio/ws_fileioservice.cpp [hunk 1]
```diff
@@ -25,106 +25,90 @@
...
+    {
+        if (stricmp(targetDZNameOrAddress, "localhost")==0)
+            targetDZNameOrAddress = ".";
 
-    Owned<IConstEnvironment> env;
-    env.setown(factory->openEnvironment());
+        Owned<IConstDropZoneInfoIterator> dropZoneItr = env->getDropZoneIteratorByAddress(targetDZNameOrAddress);
+        ForEach(*dropZoneItr)
+        {
+            IConstDropZoneInfo & dz = dropZoneItr->query();
```
Comment (wangkx) [inline esp/services/ws_fileio/ws_fileioservice.cpp]: Remove the space before the dz.

File: esp/services/ws_fileio/ws_fileioservice.cpp [hunk 2]
```diff
@@ -25,106 +25,90 @@
...
                     }
                 }
             }
+            else
+            {
+                Owned<IConstDropZoneServerInfoIterator> serverIter = dropZoneInfo->getServers();
+                ForEach(*serverIter)
+                {
+                    IConstDropZoneServerInfo &serverElem = serverIter->query();
+                    serverElem.getServer(netAddr.clear());
```
Comment (wangkx) [inline esp/services/ws_fileio/ws_fileioservice.cpp]: No clear() is needed here.

File: esp/services/ws_fileio/ws_fileioservice.cpp [hunk 3]
```diff
@@ -25,106 +25,90 @@
...
-                if (addr && *addr)
-                {           
-                    StringBuffer sNetAddr;
-                    if (strcmp(addr, "."))
-                    {       
-                        sNetAddr.append(addr);
-                    }
-                    else
+                Owned<IConstMachineInfo> machine = env->getMachine(computerName.str());
+                if (machine)
```
Comment (wangkx) [inline esp/services/ws_fileio/ws_fileioservice.cpp]: Do you need to add a log (throw an exception) if this check fails?

File: esp/services/ws_fileio/ws_fileioservice.cpp [hunk 4]
```diff
@@ -25,106 +25,90 @@
...
+                {
+                    machine->getNetAddress(computerAddress);
+                    if (computerAddress.length() != 0)
```
Comment (wangkx) [inline esp/services/ws_fileio/ws_fileioservice.cpp]: Same as line 70.

**Review by wangkx — COMMENTED** [review]
@rpastrana a few minor comments.

**Pull request merged**